### ISSUE TITLE
setMyLocationButtonEnabled not available on @ionic-native/google-maps

### ISSUE THREAD

**Issue by amsimoes** [mapsplugin/cordova-plugin-googlemaps#2122]
**I'm submitting a ...**  (check one with "x")
- [ ] question
- [ ] any problem or bug report
- [x] feature request

**If you choose 'problem or bug report', please select OS:**  (check one with "x")
- [ ] **Android**
- [ ] **iOS**

**cordova information:**  (run `$> cordova plugin list`) 

```
com.googlemaps.ios 2.6.0 "Google Maps SDK for iOS"
cordova-plugin-device 2.0.1 "Device"
cordova-plugin-googlemaps 2.2.5 "cordova-plugin-googlemaps"
cordova-plugin-ionic-keyboard 2.0.5 "cordova-plugin-ionic-keyboard"
cordova-plugin-ionic-webview 1.1.16 "cordova-plugin-ionic-webview"
cordova-plugin-splashscreen 5.0.2 "Splashscreen"
cordova-plugin-whitelist 1.3.3 "Whitelist"
```

**If you use `@ionic-native/google-maps`, please tell the package.json (only `@ionic-native/core` and `@ionic-native/google-maps` are fine mostly)**

**Current behavior:**

setMyLocationButtonEnabled instance member is not found on @ionic-native/google-maps package. I'd like to know if there's a workaround to make the button appear, or I'll just have to wait for you to implement it (I can also help you doing it if you guide me).

**Expected behavior:**

Function found and button shown, as the [ionic framework documentation](https://ionicframework.com/docs/native/google-maps/#setMyLocationButtonEnabled) says.

**Screen capture or video record:**

![error img](https://i.imgur.com/mPzgyq6.png)

**Related code, data or error log (please format your code or data):**

```
The person who share your project files on Github (or other git repository) is in faster lane than other people.
Please share your project files on Github or others(Bitbucket, Gitlabs...etc).
If you don't want to share your project files, please create a demo project, then share it.

Screen captures, and/or native logs(such as Logcat, xcode logs) are appreciate.

Giving much information, you are waiting time is less.
Thank you for your cooperation.
```

```
this.map.setMyLocationButtonEnabled(true) // undefined ...
```

**Comment by wf9a5m75** [comment]
Yes, not available yet. I have been waiting the ionic team releases the next version.

**Comment by amsimoes** [comment]
@wf9a5m75 I saw in a PR from 6 days ago, in the main ionic-native repo, that this instance has been implemented. But this plugin hasn't been yet, is it? 

If I install the npm package directly from the github repo, would it work?
What's the ETA on ionic team releasing next version?

And this next version you're speaking, is the npm version?

**Comment by wf9a5m75** [comment]
Please ask to the ionic team.

**Comment by battika** [comment]
In the meantime you can add this parameter manually to the ionic-native file.

**Comment by wf9a5m75** [comment]
@ihadeed ping

**Comment by wf9a5m75** [comment]
This is just memo: https://github.com/ionic-team/ionic-native/pull/2320

We have been just waiting the next release.
(But I don't support the capacitor)

**Comment by amsimoes** [comment]
@battika how so? Doing that would make it work?
@wf9a5m75 what do you mean by memo and capacitor?

**Comment by wf9a5m75** [comment]
The code has been already merged into the master branch, but the next version is not released yet.

**Comment by battika** [comment]
Go to your `node_modules/@ionic-native/google-maps` folder, open the file called `index.d.ts` and locate the line that reads (approx. at line 56):
`export interface GoogleMapOptions {`
Add a new line there with the following text:
`        myLocation?: boolean;
`
And voila, it works!

By the way capacitor is a new project of the Ionic team, an attempt to provide native plugins with better quality by replacing Cordova, but still backward-compatible with Cordova.
More here:
https://capacitor.ionicframework.com/docs/
and here:
https://github.com/ionic-team/capacitor

I'm really wondering how it will turn out to be, my guess is change won't happen overnight.

**Comment by amsimoes** [comment]
@wf9a5m75 So if I install the package by using the github repo, I'll get it working right?

@battika Thank you very much, I'll try it now and report asap.

**Comment by wf9a5m75** [comment]
@amsimoes You can not install the package from the git repo. You need to build the package.

The easiest solution is @battika way. Some people wants to more, they builds by themselves.

If you want to build the package, you can do like this:

```bash
$> https://github.com/ionic-team/ionic-native

$> cd ionic-native

$> npm install

// (build way #A) If you use other plugins, you need to build all of them
$> npm run build

// (build way #B) If you use only google-maps, you can build only the google-maps module
$> npm run build:core && npm run build:modules google-maps

$> cd  (your project)

$> npm link (path to)ionic-native/dist/\@ionic-native/core
$> npm link (path to)ionic-native/dist/\@ionic-native/google-maps
```

**Comment by wf9a5m75** [comment]
@battika The capacitor declares "keep the compatibility with Cordova", but they refused to implement the same installation process as cordova.
Because of this, this plugin can not declare `<meta-data android:name="com.google.android.geo.API_KEY" android:value="(your api key)" />` in the AndroidManifest.xml automatically, then app will crash easily.
I was amazed by their reaction, I won't support the capacitor.

So the memo mentions even the `ionic-native/google-maps` is updated, the maps plugin does not work on the capacitor, but I won't support the capacitor.

Anyway, the capacitor is not related to this thread directly. Just ignore it.

**Comment by battika** [comment]
@wf9a5m75 thanks for the insight. In my opinion Google Maps is a very important part of the mobile ecosystem so Capacitor will need to support it sooner or later otherwise people will refuse using it. I'm happy with your Google Maps plugin and I don't consider switching to anything else.

**Comment by wf9a5m75** [comment]
@amsimoes The ionic team finally released new version to npm.
You need to reinstall `@ionic-native/core@4.6.0` and `@ionic-native/google-maps@4.6.0`

**Comment by syedjaffersathik** [comment]
Hi, 

It is working in Android but whereas, It is not working with iOS. Kindly provide solution.

Thanks
